### DISCUSSION TITLE
솔루션 사용 시작 승인 API 호출 시 INVALID_SELLER_TOKEN 오류 리턴

### DISCUSSION THREAD

**Discussion by wileyday** [commerce-api-naver/commerce-solutions#89] [묻고 답하기] [answered]
안녕하세요?

솔루션 사용 시작 승인 API를 호출하면 INVALID_SELLER_TOKEN 오류가 리턴됩니다.
그래서 SELLER 토큰으로도 호출해 보았지만 권한이 없다는 오류가 리턴됩니다.

아래 상세 내용 남깁니다.

호출 API URL:
https://api.commerce.naver.com/external/v1/commerce-solutions/subscriptions/approve

Bearer 토큰에 입력한 토큰 타입을 SELF 로 진행하면 아래와 같이 리턴되며,

{
  code: 'INVALID_SELLER_TOKEN',
  message: 'JWe decrypt exception',
  timestamp: '2024-12-13T12:48:55.148+09:00',
  traceId: 'pr1-185309-69bxbz^1733910841346^16049820'
}

SELLER로 진행하면 아래와 같이 리턴됩니다.

{
  code: 'GW.AUTHN',
  message: '요청을 보낼 권한이 없습니다.',
  timestamp: '2024-12-13T12:52:03.548+09:00',
  traceId: 'pr1-185309-69bxbz^1733910841346^16073655'
}

Query 파라메터 token값은 별다른 설명이 없어서 넘겨받은 JWT 토큰을 그대로 입력하였습니다.

감사합니다.

**Comment by commerce-api-naver** [comment, marked as answer]
안녕하세요. 커머스API 이민석입니다.

> [!IMPORTANT]
> 게시해주신 내용은 커머스솔루션 개발사 연동 관련 문의로 판단되어 관리자가 게시 공간을 이동하였습니다.
> - 최초 게시된 공간: 커머스API 내 스토어 애플리케이션 기술지원 공간 (`/commerce-api`)
> - 이동된 공간: 커머스솔루션 개발사 기술지원 공간 (`/commerce-solutions`)

### API 설명
'[솔루션] 사용 시작 승인 (`PUT /v1/commerce-solutions/subscriptions/approve`)' API는...
- 솔루션 애플리케이션의 SELF 인증 토큰을 사용합니다.
- 심사 후 승인형 솔루션만 사용이 가능합니다. (별도 설정 및 '판매자 커머스ID 인증' 개발 필요)
    - 관련한 도움말: [커머스API센터 > 솔루션 개발 가이드 > 심사 후 승인 옵션 사용 안내](https://apicenter.commerce.naver.com/ko/basic/solution-doc#section/%EA%B0%9C%EB%B0%9C%ED%95%98%EA%B8%B0/%EC%8B%AC%EC%82%AC-%ED%9B%84-%EC%8A%B9%EC%9D%B8-%EC%98%B5%EC%85%98-%EC%82%AC%EC%9A%A9-%EC%95%88%EB%82%B4)

### 문의 내용에 대한 답변
[Trace ID: pr1-185309-69bxbz^1733910841346^16049820]에 해당하는 호출건에서 `token` 파라미터로 전달해주신 내용은 (대상 API에서 요구하는 **JWE 토큰**이 아닌) **JWT 토큰**입니다.
JWT 토큰은 모든 커머스솔루션에서 기본으로 제공되는 항목으로 대상 토큰에 대한 정보 획득은 API 호출이 아닌 JWT 복호화로 API 사용자(솔루션) 시스템 로컬에서 수행합니다.
- 관련한 도움말: [커머스API센터 > 솔루션 개발 가이드 > 기본 연동 요소 가이드](https://apicenter.commerce.naver.com/ko/basic/solution-doc#section/%EA%B0%9C%EB%B0%9C%ED%95%98%EA%B8%B0/%EA%B8%B0%EB%B3%B8-%EC%97%B0%EB%8F%99-%EC%9A%94%EC%86%8C-%EA%B0%80%EC%9D%B4%EB%93%9C)

이용하시려는 API([솔루션] 사용 시작 승인)은 심사 후 승인형 솔루션에서 구독자(사용자)가 판매자 커머스ID 인증이 완료된 뒤 다시 솔루션 웹 사이트로 복귀하는 트래픽에서 전달되는 JWE 토큰 값을 활용합니다.

만일, 운용하시려는 솔루션이 별도의 가입 심사 등이 필요치 않아 커머스솔루션마켓에서 구독 즉시 이용이 가능한 유형이라면 상기로 알려드린 '솔루션 개발 가이드' 내용을 참고하시어 JWT 복호화로 필요 정보를 획득하시기 바랍니다.
(단, 복호화 시 제공 토큰에 대한 검증이 필요합니다. 상세 내용은 개발 가이드를 참고해주시기 바랍니다.)

그렇지 않고 별도의 가입 과정 등이 동반되어 커머스솔루션마켓에서의 구독 절차와는 별개로 솔루션 시스템에서의 가입/인증 절차가 필요하다면 (1)커머스API센터에서 대상 솔루션에 대한 '심사 후 승인형' 설정 후, (2)판매자 커머스ID 인증을 구현하여 복귀 URL(Redirect URL)로 수신받는 JWE 토큰 항목을 대상 API 호출에 사용해보시기 바랍니다.

**Reply by wileyday** [reply]
별도의 승인 절차가 필요치 않은 솔루션이라서 "솔루션 사용 시작 승인 API"를 호출할 필요가 없군요!
감사합니다^^
